docs/labels.qmd

```diff
@@ -0,0 +1,403 @@
+
+-   **No Interference with tagged NAs**: Operations on labels will not remove or corrupt tagged NA values. For instance, if a variable contains NA(b) to indicate some missing category, setting or getting labels for that variable should not turn NA(b) into a regular NA or a numeric code. The underlying data type remains the same. Our implementation will recognize `haven`’s tagged NA class and handle it accordingly. If we summarize or print labeled data, we will rely on haven/labelled print methods that already know how to display tagged NAs (showing NA(b) etc.) ￼. If any custom processing is done, the NA status will be preserved. In essence, tagged NAs will be treated as a valid “value” in the value labels vector (so they can have a label), but we won’t inadvertently treat them as ordinary values.
+
+-   **Clear specification for missing in metadata:** The specification will clarify how to denote missing values in `variables.csv` or `variable_details.csv`. For example, we may decide that within the metadata, a certain column (or perhaps a special notation in the labels) indicates missing. Some systems use an annotation like “” next to the label. We might introduce a convention such as: if a value label is enclosed in brackets or flagged in an additional column (e.g., a boolean missing column), `recodeflow` will know to handle that code as missing. This detail will be ironed out in the tagged NA feature, but the label functionality will be aware of it. The result should be that if a user marks a code as missing in the metadata, after applying labels, the R data’s values for that code are converted to NA (with tag if needed) and carry the appropriate label. For SPSS-like user missing values, we will consider using the class `haven`\_`labelled`\_spss which allows storing a separate na_values attribute for truly round-trip compatibility ￼, but even if not, the combination of tagged NA and label will serve the purpose.
```

Comment:
The sheets already have a way of representing tagged NA values by using NA::a, NA::b. Were you thinking of changing that?